src/data_loader.py

```diff
@@ -0,0 +1,160 @@
+import os
+import PIL
+import torch
+import random
+import collections
+import numpy as np
+import pandas as pd
+
+
+
+class ImageSegmentaionDataset():
+    """Image Segmentation dataset"""
+    
+    def __init__(self, image_dict, loader_type, replacement=True, transform=None):
+        """
+        Args:
+        image_dict (dict): a dict of the following format:
+                        {"ships": {"imageId": "segmentation pixels"}， “w/ships”: ["imageId"]}
+        loader_type: (string)"classification" or "segmentation"
+        replacement: (bool) sample with/without replacement
+        transform (callable, optional): Optional transform to be applied on a sample.
+        """
+        self.image_dict = image_dict
```

Comment:
Could we use a bit more descriptive naming? Also, could we not use "dict" since it's an [anti-pattern](https://www.codewithjason.com/variable-name-anti-patterns/)? (see "Naming a Variable after Its Type"

Reply:
I think I will change that later if I had time because I cannot think of another appropriate name now......